app/master/build.py

```diff
@@ -307,6 +220,10 @@ def _mark_subjob_complete(self, subjob_id):
             self._logger.info("All results received for build {}!", self._build_id)
             SafeThread(target=self._perform_async_postbuild_tasks, name='PostBuild{}'.format(self._build_id)).start()
 
+    def mark_started(self):
+        self._setup_is_started = True
+        self._record_state_timestamp(BuildStatus.BUILDING)
```

Comment:
Just for reference, the `_record_state_timestamp` correspond pretty closely to where the state machine will be making transitions.